test/testinfra.py

```diff
@@ -255,20 +255,27 @@ def get(self, resource):
             self.cache(resource, response)
             return json.loads(response['data'])
 
-    def post(self, resource, data):
+    def post(self, resource, data, accept=[]):
         response = self.request("POST", resource, json.dumps(data), { "Content-Type": "application/json" })
-        if response['status'] < 200 or response['status'] >= 300:
-            sys.stderr.write(response['data'])
+        status = response['status']
+        if (status < 200 or status >= 300) and status not in accept:
+            sys.stderr.write("{0}\n{1}\n", resource, response[data])
             raise Exception("GitHub API problem: {0}".format(response['reason'] or response['status']))
         return json.loads(response['data'])
 
     def statuses(self, revision):
         result = { }
-        statuses = self.get("commits/{0}/statuses".format(revision))
-        if statuses:
-            for status in statuses:
-                if status["context"] not in result:
-                    result[status["context"]] = status
+        page = 1
+        count = 100
+        while count == 100:
+            statuses = self.get("commits/{0}/statuses?page={1}&per_page={2}".format(revision, page, count))
+            count = 0
+            page += 1
+            if statuses:
+                for status in statuses:
+                    count += 1
```

Comment:
instead of setting `count` to 0 and incrementing by 1, can't we just set it to the length of `statuses` right away?